### PR TITLE
Design of empty tensor and slice required for implementing if/else/switch op

### PR DESCRIPTION
**Pull request by zchen0211**
https://github.com/zchen0211/Paddle/blob/design/doc/design/empty_tensor_slice.md

### CODE REVIEW

File: paddle/operators/scatter.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
```
Comment (wangkuiyi) [inline paddle/operators/scatter.h]: It looks like most content of this PR duplicates with https://github.com/PaddlePaddle/Paddle/pull/3371

File: doc/design/empty_tensor_slice.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+### Empty Tensor Design
```
Comment (wangkuiyi) [inline doc/design/empty_tensor_slice.md]: I think we don't need to create a new .md file here; instead, we can move the content here to https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/tensor.md

Comment (zchen0211) [inline doc/design/empty_tensor_slice.md]: I will move it...

File: paddle/operators/scatter.h (hunk 1 above)
Comment (zchen0211) [inline paddle/operators/scatter.h]: Sure. Thanks.

**Pull request closed without merging**